omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderFrequencyResource1_10.java

```diff
@@ -68,6 +69,30 @@ public DelegatingResourceDescription getRepresentationDescription(Representation
 		return null;
 	}
 	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		
+		description.addRequiredProperty("frequencyPerDay");
+		description.addRequiredProperty("concept");
+		
+		return description;
+	}
+	
+	/**
+	 * Annotated setter for Concept
+	 * 
+	 * @param orderFrequency
+	 * @param value
+	 */
+	@PropertySetter("concept")
+	public static void setConcept(OrderFrequency orderFrequency, Object value) {
```

Comment:
What is the use of this method?

Reply:
This is described in the comment @dkayiwa . If this requires documentation, the whole resource needs it and I thought of creating a curated intro ticket for this. Otherwise [these](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/OrderFrequency.java#L25-L29) are the minimum required properties to be set in order to GET or POST an `OrderFrequency` .

Reply:
What happens if you do not have this method?

Reply:
The root cause for including this method and not one for the `frequencyPerDay` is [here](https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/BaseDelegatingResource.java#L755-L791). What happens if I do not include this method is, there is no way the string Uuid (mapped with the `concept` property in the `JSON` object) referencing the Concept in question (must exit in the database) can be converted to the Concept object which is a field in the `OrderFrequency` object that is to be persisted. I stand to be corrected.

Reply:
Do you have a unit test which fails if this method is not included?

Reply:
I actually didn't  think of this magic. No unit test fails. Let me investigate this and refactor accordingly. Thank you @dkayiwa.